plugins/content/bower/index.js

```diff
@@ -779,30 +779,33 @@ function handleUploadedPlugin (req, res, next) {
     }
 
     // try unzipping
+
     var outputPath = file.path + '_unzipped';
-    var rs = fs.createReadStream(file.path);
-    var ws = unzip.Extract({ path: outputPath });
-    rs.on('error', function (error) {
-      return next(error);
-    });
-    ws.on('error', function (error) {
+
+    fs.createReadStream(file.path)
+    .pipe(unzip.Extract({ path: outputPath }))
+    .on('error', function (error) {
       return next(error);
-    });
-    ws.on('close', function () {
+    })
+    .on('close', function (data) {
       // enumerate output directory and search for bower.json
       fs.readdir(outputPath, function (err, files) {
-        if (err) {
-          return next(err);
-        }
+        if (err) return next(err);
```

Comment:
I'm not a huge fan of this formatting style, although we've used it in the frontend. I prefer my curlies.